src/browser/transports/injection_bus.ts

```diff
@@ -88,28 +91,84 @@ export default class NativeWindowInjectionBus extends EventEmitter {
         return;
       }
 
-      // Ack / Nack
+      this._listenerHeartbeat(parsedMessage);
+
       if (this._pendingRequests.has(messageId)) {
-        const isNack = parsedMessage.action.includes('error');
-        const pendingRequest = this._pendingRequests.get(messageId);
+        this._listenerAck(parsedMessage);
+      } else {
+        this._listenerBroadcastMsg(parsedMessage);
+      }
+    };
 
-        isNack
-          ? pendingRequest.reject((<NackMessage>parsedMessage).payload.reason)
-          : pendingRequest.resolve();
+    copyDataTransport.on('message', this._messageListener);
+
+    this.subscribe();
+  }
 
-        this._senderId = senderId;
-        this._pendingRequests.delete(messageId);
+  // Setup heartbeat
+  private setupHeartbeat() {
+    this._connected = false;
+    this._previousDllSequence = -1;
 
-        return;
+    this._listenerHeartbeat = (message: MessageBase): void => {
+      const { sequence } = message;
+
+      if (this._previousDllSequence + 1 !== sequence) {
```

Comment:
1. After this is off once will it ever self correct or just log that it was missed over and over?
2. Should it raise and event or tell the other party that it was missed?
3. Is this ever meant to be user facing or is this a protocol level thing?

Reply:
This is a protocol thing for now. Added this minimal logic here, because we still don't know what the behavior in this case should be. Technically, we shouldn't have any missed message as the sender knows when something fails to be send synchronously. This is just initial check, which will only log. Once and if the real-world problems start to arise for this case, we'll have to figure out how we will handle them.

Reply:
added re-resubscription in this case